### PR TITLE
Changes to test configurations options and externals for parsing

### PR DESCRIPTION
**Pull request by fmalatino**
**Description**
Conditional statement in `stencils/testing/test_translate.py` changed to check for `compute` passed via the `--grid` pytest option.

Switched conditional statement in `ndsl/stencils/testing/conftest.py` to check for option of `cubed-sphere` selection

Made `mysign` parameter of `fill_corners_dgrid_defn` an external variable

Resolves Issue [52](https://github.com/NOAA-GFDL/NDSL/issues/52)

**How Has This Been Tested?**
Tested in `pyFV3` and `pySHiELD` translate tests github workflow.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included

### CODE REVIEW

File: ndsl/stencils/testing/conftest.py [hunk 1]
```diff
@@ -370,7 +370,7 @@ def generate_parallel_stencil_tests(metafunc, *, backend: str):
 
 
 def get_communicator(comm, layout, topology_mode):
-    if (MPI.COMM_WORLD.Get_size() > 1) and (topology_mode == "doubly-periodic"):
+    if (MPI.COMM_WORLD.Get_size() > 1) and (topology_mode == "cubed-sphere"):
```
Comment (bensonr) [inline ndsl/stencils/testing/conftest.py:373]: This change will definitely make a big difference!!

**Review by bensonr — DISMISSED** [review]

**Review by FlorianDeconinck — CHANGES_REQUESTED** [review]
Revert the `mysign`
Revert the `gt4py` and `dace` changes it's out of PR

**Review by FlorianDeconinck — APPROVED** [review]
LGTM.

This will have downstream effect on `pyFV3` due to the external change. The `mysign` change has been approved by me. It's only used like this and therefore should be explicit .

**Review by oelbert — APPROVED** [review]
LGTM. Can you put an issue in PyFV3 to propagate the change to `fill_corners_dgrid_defn` there once this is merged?

**Comment by fmalatino** [comment]
> LGTM. Can you put an issue in PyFV3 to propagate the change to `fill_corners_dgrid_defn` there once this is merged?

@oelbert those changes should be currently in [PR 23](https://github.com/NOAA-GFDL/PyFV3/pull/23), which is still in draft as it is dependent on this PR.

**Pull request merged**